File: apps/myjobhunter/backend/app/schemas/profile/screening_answer.py
"""ScreeningAnswer schemas — Phase 1 stub."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class ScreeningAnswerRead(BaseModel):
    id: uuid.UUID
    profile_id: uuid.UUID
    question_key: str
    answer: str | None
    is_eeoc: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/myjobhunter/frontend/src/types/screening-answer/screening-answer.ts
/**
 * ScreeningAnswer as returned by the backend.
 * Mirrors ScreeningAnswerResponse in backend/app/schemas/profile/screening_answer_response.py.
 *
 * Note: is_eeoc is server-derived from question_key. Never send it in requests.
 */
export interface ScreeningAnswer {
  id: string;
  user_id: string;
  profile_id: string;
  question_key: string;
  answer: string | null;
  is_eeoc: boolean;
  created_at: string;
  updated_at: string;
}
